pkg/api/validation/validation.go

```diff
@@ -2844,10 +2799,16 @@ func ValidateNodeUpdate(node, oldNode *api.Node) field.ErrorList {
 	// Clear status
 	oldNode.Status = node.Status
 
+	// update taints
```

Comment:
@liggitt I think we may need to bite the bullet and change all the admission controllers to accommodate it. I think it's not possible for Kubernetes or a cluster admin to roll out new taints in a non-disruptive way if we don't allow updates to tolerations on existing pods.

For example, say a new Kubelet release introduce a new "node problem" NoExecute taint, and by default pods should tolerate it for 5 minutes. We can add an admission controller to add the default toleration, but the admin also needs to update the pods that are pending and running at the time they upgrade the Kubelet, otherwise if any of their nodes get the taint then the pods that were pending or running will be evicted immediately which is not what they want. 

Another example would be, you convert your cluster from shared to partitioned by team by introducing NoSchedule taints to all of the nodes. If you do this, all of the pods that are pending at the time you add the taints will remain pending forever (since none of them tolerate the dedicated node taints). What you want is to update the pending pods to tolerate the taint corresponding to their dedicated node group.

If you think it is not possible to update the security assumptions of your admission controllers in time for 1.6, we could say toleration update is rejected in 1.6 and then allowed in 1.7 (which would be considered a backward-compatible change IMO). But this feature is much less useful without it, so I don't see how we can avoid allowing toleration update either now or very soon.